pyriemann_qiskit/classification.py

```diff
@@ -0,0 +1,318 @@
+
+    Attributes
+    ----------
+    classes_ : list
+        list of classes.
+    verbose : see above
+    processVector : see above
+    qAccountToken : see above
+    target : see above
+    quantum : see above
+    test_input : Dictionnary
+        Contains vectorized test set for target and non-target classes
+    training_input : Dictionnary
+        Contains vectorized training set for target and non-target classes
+    provider : IBMQ Provider
+        This service provide a remote quantum computer backend
+    backend : Quantum computer or simulator
+    feature_dim : int
+        Size of the vectorized matrix which is passed to quantum classifier
+    new_feature_dim : int
+        Feature dimension after proccessed by `processVector` lambda
+    prev_fit_params : Dictionnary of data and labels
+        Keep in memory data and labels passed to fit method.
+        This is used for self-calibration.
+    feature_map: ZZFeatureMap
+        Transform data into quantum space
+    quantum_instance: QuantumInstance (Object)
+        Backend with specific parameters (number of shots, etc.)
+
+    See Also
+    --------
+    QuanticSVM
+    QuanticVQC
+
+    """
+
+    def __init__(self, target, qAccountToken=None, quantum=True,
+                 processVector=lambda v: v, verbose=True, **parameters):
+        self.verbose = verbose
+        self.log("Initializing Quantum Classifier")
+        self.test_input = {}
+        self.set_params(**parameters)
+        self.processVector = processVector
+        self.qAccountToken = qAccountToken
+        self.training_input = {}
+        self.target = target
+        self.quantum = quantum
+        if quantum:
+            aqua_globals.random_seed = datetime.now().microsecond
+            self.log("seed = ", aqua_globals.random_seed)
+            if qAccountToken:
+                self.log("Real quantum computation will be performed")
+                IBMQ.delete_account()
+                IBMQ.save_account(qAccountToken)
+                IBMQ.load_account()
+                self.log("Getting provider...")
+                self.provider = IBMQ.get_provider(hub='ibm-q')
+            else:
+                self.log("Quantum simulation will be performed")
+                self.backend = BasicAer.get_backend('qasm_simulator')
+        else:
+            self.log("Classical SVM will be performed")
+
+    def log(self, *values):
+        if self.verbose:
+            print("[QClass] ", *values)
+
+    def vectorize(self, X):
+        vector = X.reshape(len(X), self.feature_dim)
+        return [self.processVector(x) for x in vector]
+
+    def splitTargetAndNonTarget(self, X, y):
+        self.log("""[Warning] Spitting target from non target.
+                 Only binary classification is supported.""")
+        nbSensor = len(X[0])
+        try:
+            nbSamples = len(X[0][0])
+        except Exception:
+            nbSamples = 1
+        self.feature_dim = nbSensor * nbSamples
```

Comment:
To be homogeneous with pyRiemann variable naming, can you use `n_channels` instead of `nbSensor`, and `n_matrices` instead of `nbSamples` ?

See
https://github.com/pyRiemann/pyRiemann/pull/120/files#r646880031
and
https://github.com/pyRiemann/pyRiemann/pull/140#discussion_r714660528